scripts/globals/abilities/full_circle.lua

```diff
@@ -0,0 +1,36 @@
+-----------------------------------
+-- Ability: Full Circle
+-- Causes your luopan to vanish.
+-- Obtained: Geomancer Level 5
+-- Recast Time: 10 seconds
+-- Refunds some of the MP consumed by the Geocolure spell that created the luopan.
+-- Amount of MP restored varies depending on remaining Luopan HP.
+-----------------------------------
+require("scripts/globals/job_utils/geomancer")
+require("scripts/globals/ability")
+require("scripts/globals/status")
+require("scripts/settings/main")
+require("scripts/globals/msg")
+-----------------------------------
+local ability_object = {}
+
+ability_object.onAbilityCheck = function(player, target, ability)
+    return xi.job_utils.geomancer.geoOnAbilityCheck(player, target, ability)
+end
+
+ability_object.onUseAbility = function(player, target, ability)
+    local luopan        = player:getPet()
+    local hpp_remaining = luopan:getHPP()
+    local mp_cost       = luopan:getLocalVar("MP_COST")
+    local fcMerit       = player:getMerit(xi.merit.FULL_CIRCLE_EFFECT)
+    local mp_returned   = ((0.5 + (fcMerit /100)) *mp_cost) *hpp_remaining / 100
```

Comment:
Order of operations is unclear:
` ((0.5 + (fcMerit / 100)) * mp_cost) * (hpp_remaining / 100)`

Reply:
Should be a bit simpler to follow now